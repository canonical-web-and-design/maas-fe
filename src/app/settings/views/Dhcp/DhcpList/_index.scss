@mixin DhcpList {
  .dhcp-list td,
  .dhcp-list th {
    // Snippet name:
    &:nth-child(1) {
      flex: 0.2 0 0;
    }

    // Type
    &:nth-child(2) {
      flex: 0.08 0 0;
    }

    // Applies to
    &:nth-child(3) {
      flex: 0.12 0 0;
    }

    // Description
    &:nth-child(4) {
      flex: 0.3 0 0;
    }

    // Enabled
    &:nth-child(5) {
      flex: 0.08 0 0;
    }

    // Last edited
    &:nth-child(6) {
      flex: 0.14 0 0;
    }

    // Actions
    &:nth-child(7) {
      flex: 0.08 0 0;
    }
  }

  @media screen and (max-width: 1050px) {
    .dhcp-list td,
    .dhcp-list th {
      // Applies to
      &:nth-child(3) {
        flex-grow: 0.2;
      }

      // Last edited
      &:nth-child(6) {
        display: none;
      }

      // Actions:
      &:nth-child(7) {
        flex-grow: 0.14;
      }
    }
  }

  @media screen and (max-width: 930px) {
    .dhcp-list td,
    .dhcp-list th {
      // Snippet name:
      &:nth-child(1) {
        flex-grow: 0.28;
      }

      // Type
      &:nth-child(2) {
        display: none;
      }
    }
  }

  @media screen and (max-width: 660px) {
    .dhcp-list td,
    .dhcp-list th {
      // Snippet name:
      &:nth-child(1) {
        flex-grow: 0.43;
      }

      // Applies to
      &:nth-child(3) {
        flex-grow: 0.35;
      }

      // Description
      &:nth-child(4) {
        display: none;
      }
    }
  }

  @media screen and (max-width: 530px) {
    .dhcp-list td,
    .dhcp-list th {
      // Snippet name:
      &:nth-child(1) {
        flex-grow: 0.68;
      }

      // Applies to
      &:nth-child(3) {
        display: none;
      }

      // Actions:
      &:nth-child(7) {
        flex-grow: 0.24;
      }
    }
  }
}
