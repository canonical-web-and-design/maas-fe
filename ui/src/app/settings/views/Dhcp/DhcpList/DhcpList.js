import { useDispatch, useSelector } from "react-redux";
import React, { useEffect } from "react";

import actions from "app/settings/actions";
import Loader from "app/base/components/Loader";
import selectors from "app/settings/selectors";

// Temporarily create a item component, this will be replaced with a table.
const DhcpListItem = ({ id, name, subnet }) => {
  const subnetItem = useSelector(state =>
    selectors.subnet.getById(state, subnet)
  );
  return (
    <li>
      {name} {subnet && `(subnet: ${subnetItem.name})`}
    </li>
  );
};

const DhcpList = () => {
  const dhcpsnippetLoading = useSelector(selectors.dhcpsnippet.loading);
  const dhcpsnippetLoaded = useSelector(selectors.dhcpsnippet.loaded);
  const dhcpsnippets = useSelector(selectors.dhcpsnippet.all);
  const subnetLoading = useSelector(selectors.subnet.loading);
  const subnetLoaded = useSelector(selectors.subnet.loaded);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(actions.dhcpsnippet.fetch());
    dispatch(actions.subnet.fetch());
  }, [dispatch]);

  if (dhcpsnippetLoading || subnetLoading) {
    return <Loader text="Loading..." />;
  }
  const snippets = dhcpsnippets.map(dhcpsnippet => (
    <DhcpListItem {...dhcpsnippet} key={dhcpsnippet.id} />
  ));
  return dhcpsnippetLoaded && subnetLoaded && <ul>{snippets}</ul>;
};

export default DhcpList;
