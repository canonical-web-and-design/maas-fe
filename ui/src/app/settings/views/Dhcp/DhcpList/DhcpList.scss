.dhcp-list td,
.dhcp-list th {
  // Snippet name:
  &:nth-child(1) {
    flex: 0 0 20%;
  }

  // Type
  &:nth-child(2) {
    flex: 0 0 8%;
  }

  // Applies to
  &:nth-child(3) {
    flex: 0 0 12%;
  }

  // Description
  &:nth-child(4) {
    flex: 0 0 30%;
  }

  // Enabled
  &:nth-child(5) {
    flex: 0 0 8%;
  }

  // Last edited
  &:nth-child(6) {
    flex: 0 0 14%;
  }
}
