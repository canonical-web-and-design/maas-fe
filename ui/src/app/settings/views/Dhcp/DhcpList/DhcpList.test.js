import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import React from "react";

import DhcpList from "./DhcpList";

const mockStore = configureStore();

describe("DhcpList", () => {
  let state;

  beforeEach(() => {
    state = {
      dhcpsnippet: {
        loading: false,
        loaded: true,
        items: [{ id: 1, name: "class" }, { id: 2, name: "lease", subnet: 2 }]
      },
      subnet: {
        loading: false,
        loaded: true,
        items: [{ id: 1, name: "10.0.0.99" }, { id: 2, name: "test.maas" }]
      }
    };
  });

  it("displays a loading component if loading", () => {
    state.dhcpsnippet.loading = true;
    state.subnet.loading = true;
    const store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <DhcpList />
      </Provider>
    );
    expect(wrapper.find("Loader").exists()).toBe(true);
  });

  it("displays a list of dhcp snippets", () => {
    const store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <DhcpList />
      </Provider>
    );
    expect(wrapper.find("DhcpList")).toMatchSnapshot();
  });
});
