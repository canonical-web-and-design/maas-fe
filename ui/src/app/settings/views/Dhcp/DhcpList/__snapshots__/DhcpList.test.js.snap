// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DhcpList displays a list of dhcp snippets 1`] = `
<DhcpList>
  <ul>
    <DhcpListItem
      id={1}
      key="1"
      name="class"
    >
      <li>
        class
         
      </li>
    </DhcpListItem>
    <DhcpListItem
      id={2}
      key="2"
      name="lease"
      subnet={2}
    >
      <li>
        lease
         
        (subnet: test.maas)
      </li>
    </DhcpListItem>
  </ul>
</DhcpList>
`;
